src/test/java/uk/gov/hmcts/reform/ccd/service/CaseFinderServiceTest.java

```diff
@@ -0,0 +1,248 @@
+import static uk.gov.hmcts.reform.ccd.fixture.TestData.LINKED_CASE_DATA_101;
+import static uk.gov.hmcts.reform.ccd.fixture.TestData.LINKED_CASE_DATA_11;
+import static uk.gov.hmcts.reform.ccd.fixture.TestData.NON_DELETABLE_CASE_WITH_TODAY_TTL;
+
+@ExtendWith(MockitoExtension.class)
+class CaseFinderServiceTest {
+    @Mock
+    private CaseDataRepository caseDataRepository;
+    @Mock
+    private CaseLinkRepository caseLinkRepository;
+    @Mock
+    private ApplicationParameters parameters;
+    @InjectMocks
+    private CaseFinderService underTest;
+
+    private final List<String> deletableCaseTypes = List.of(DELETABLE_CASE_TYPE);
+
+    @Test
+    void testGetExpiredCases() {
+        final List<CaseDataEntity> expiredCases = List.of(DELETABLE_CASE_WITH_PAST_TTL);
+        doReturn(deletableCaseTypes).when(parameters).getDeletableCaseTypes();
+        doReturn(expiredCases).when(caseDataRepository).findExpiredCases(deletableCaseTypes);
+
+        final List<CaseDataEntity> actualExpiredCases = underTest.getExpiredCases();
+
+        assertThat(actualExpiredCases)
+            .isNotEmpty()
+            .hasSize(1);
+        verify(parameters).getDeletableCaseTypes();
+        verify(caseDataRepository).findExpiredCases(deletableCaseTypes);
+    }
+
+    @Test
+    void testGetLinkedCases() {
+        final List<CaseLinkEntity> caseLinks = List.of(
+            new CaseLinkEntityBuilder(10L, DELETABLE_CASE_TYPE, DELETABLE_CASE_WITH_PAST_TTL.getId()).build(),
+            new CaseLinkEntityBuilder(11L, DELETABLE_CASE_TYPE, DELETABLE_CASE_WITH_PAST_TTL.getId()).build()
+        );
+        final List<CaseDataEntity> expectedLinkedCases = List.of(LINKED_CASE_DATA_10, LINKED_CASE_DATA_11);
+        doReturn(caseLinks).when(caseLinkRepository).findByCaseId(DELETABLE_CASE_WITH_PAST_TTL.getId());
+        doReturn(expectedLinkedCases).when(caseDataRepository).findAllById(anyList());
+
+        final List<CaseDataEntity> linkedCases = underTest.getLinkedCases(DELETABLE_CASE_WITH_PAST_TTL);
+
+        assertThat(linkedCases)
+            .isNotEmpty()
+            .hasSameElementsAs(expectedLinkedCases);
+
+        verify(caseLinkRepository).findByCaseId(DELETABLE_CASE_WITH_PAST_TTL.getId());
+        verify(caseDataRepository).findAllById(anyList());
+    }
+
+    @Test
+    void testGetLinkedCasesWhenNoCasesAreLinked() {
+        doReturn(emptyList()).when(caseLinkRepository).findByCaseId(DELETABLE_CASE_WITH_PAST_TTL.getId());
+        doReturn(emptyList()).when(caseDataRepository).findAllById(anyList());
+
+        final List<CaseDataEntity> linkedCases = underTest.getLinkedCases(DELETABLE_CASE_WITH_PAST_TTL);
+
+        assertThat(linkedCases)
+            .isEmpty();
+
+        verify(caseLinkRepository).findByCaseId(DELETABLE_CASE_WITH_PAST_TTL.getId());
+        verify(caseDataRepository).findAllById(emptyList());
+    }
+
+    @Test
+    void testIsCaseDueDeletionWhenCaseIsOfDeletableTypeWithTtlInThePast() {
```

Comment:
Is there scope to use parameterized tests here many of these similar looking tests?